frontend/src/components/CourseModal/CourseModal.tsx

```diff
@@ -181,6 +183,20 @@ function CourseModal() {
       setHistory([listingFromQuery]);
     });
   }, [history.length, searchParams, requestSeasons, courses]);
+  useEffect(() => {
+    const crossSections: CourseModalHeaderData[] = [];
+    const courseModal = searchParams.get('course-modal');
+    if (!courseModal) return;
+    const seasonCode = courseModal.split('-')[0] as Season;
+    courses[seasonCode]?.forEach((course: CourseModalHeaderData) => {
```

Comment:
@wesleyandrade05 So this may be a bit confusing, but `courses[seasonCode]` does not always exist. It only exists if the course modal is opened from the catalog, or otherwise the corresponding JSON happens to be available because the user has viewed the catalog/worksheet of that semester. If you navigate to this modal using the past course links, then the full catalog will not exist because this modal data is acquired from GraphQL not from the static JSON catalog. Therefore I think you should always request fresh data from GraphQL instead.

Reply:
You mean using the `requestSeasons` function?

Reply:
No. That one downloads the full JSON data. We just want to be able to get the related course data about all other sections. In fact all we need is the CRN.

Reply:
Got it. Where can I see what is the specific API I would need to query? I am a bit lost on that. I see this is what gets the full catalog:
```
const res = await fetchAPI(`/catalog/public/${season}`, {
    breadcrumb,
  });
```

But I can't see where I can just fetch graphQL for a specific coursename, for example (to find the sections)

Reply:
See how we fetch the course modal info for example. It's quite complicated actually and we probably benefit from some docs, but TLDR there's no separate endpoint for it. You write a GQL query in the GQL query under queries, then run codegen, and then import the generated SDK. Take a look at how the other queries there are used.